hyperspy/docstrings/model.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2020 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Common docstring snippets for model.
+
+"""
+
+FIT_PARAMETERS_ARG = """optimizer : str or None, default None
+            The optimization algorithm used to perform the fitting.
+
+                * "lm" performs least-squares optimization using the
```

Comment:
Although I see the point of dividing each bounding approach into separate optimizers, I think that this can lead to conceptual confusion. Why not using ``"lm"`` for the optimizer and then use ``bounded`` to choose between ``"trf", "dogbox" and "mpfit"?

By the way, you've made mpfit the default optimizer for bounded-LM, is that because you've performed some tests and you've found it to be superior to the others?

Reply:
But "trf" and "dogbox" aren't the Levenberg-Marquardt algorithm, so that doesn't make sense.

> mpfit the default optimizer for bounded-LM

It's the _only_ bounded Levenberg-Marquardt option.

"trf" and "dogbox" are alternative least-squares minimization algorithms, which do support bounds and (optionally) sparse Jacobians. But they are not the Levenberg-Marquardt algorithm (`optimizer="lm"`).

Reply:
You are right, the scipy's API, where you get those options in ``method`` (what I (mis)understood as bounding_method), led me to conceptual misunderstanding, so clearly your approach is clearer.